pkg/generate/app/env.go

```diff
@@ -82,3 +124,64 @@ func JoinEnvironment(a, b []kapi.EnvVar) (out []kapi.EnvVar) {
 	}
 	return out
 }
+
+func LoadEnvironmentFile(filename string, stdin io.Reader) (Environment, error) {
+	errorFilename := filename
+
+	if filename == "-" && stdin != nil {
+		//once https://github.com/joho/godotenv/pull/20 is merged we can get rid of using tempfile
+		temp, err := ioutil.TempFile("", "origin-env-stdin")
+		if err != nil {
+			return nil, fmt.Errorf("Cannot create temporary file: %s", err)
+		}
+
+		filename = temp.Name()
+		errorFilename = "stdin"
+		defer os.Remove(filename)
+
+		if _, err = io.Copy(temp, stdin); err != nil {
+			return nil, fmt.Errorf("Cannot write to temporary file %q: %s", filename, err)
+		}
+		temp.Close()
+	}
+
+	env, err := godotenv.Read(filename)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot read variables from file %s: %s", errorFilename, err)
+	}
+	return env, nil
+}
+
+// ParseAndCombineEnvironment parses key-value records from the command line
+// and from given files and combines them into single map.
+//
+// The dupfn function is called for all duplicate keys that encountered. If the
+// functions returns an error it is added to the list of errors returned by
+// ParseAndCombineEnvironment.
+//
+// If a file is "-" the file contents will be read from argument stdin (unless
+// it's nil).
+func ParseAndCombineEnvironment(cmdline []string, filenames []string, stdin io.Reader, dupfn func(string, string) error) (Environment, []error) {
+	vars, duplicates, errs := ParseEnvironment(cmdline...)
+	for _, s := range duplicates {
+		if dupErr := dupfn(s, ""); dupErr != nil {
+			errs = append(errs, dupErr)
+		}
+	}
+
+	for _, fname := range filenames {
+		fileVars, fileErr := LoadEnvironmentFile(fname, stdin)
```

Comment:
godoc should also mention that it doesn't fail right away for errors loading the provided file. In fact, it feels like it should, if I do something completely wrong like `--env-file=/nonexistent`.